buildkite/resource_cluster_agent_token.go

```diff
@@ -0,0 +1,193 @@
+			},
+			"cluster_uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (ct *ClusterAgentToken) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state ClusterAgentTokenResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Creating cluster agent token with description %s into cluster %s ...", plan.Description.ValueString(), plan.ClusterId.ValueString())
+	r, err := createClusterAgentToken(
+		ct.client.genqlient,
+		ct.client.organizationId,
+		plan.ClusterId.ValueString(),
+		plan.Description.ValueString(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create Cluster Agent Token",
+			fmt.Sprintf("Unable to create Cluster Agent Token: %s", err.Error()),
+		)
+		return
+	}
+
+	state.Id = types.StringValue(r.ClusterAgentTokenCreate.ClusterAgentToken.Id)
+	state.Uuid = types.StringValue(r.ClusterAgentTokenCreate.ClusterAgentToken.Uuid)
+	state.Description = types.StringValue(r.ClusterAgentTokenCreate.ClusterAgentToken.Description)
+	state.Token = types.StringValue(r.ClusterAgentTokenCreate.TokenValue)
+	state.ClusterId = types.StringValue(r.ClusterAgentTokenCreate.ClusterAgentToken.Cluster.Id)
+	state.ClusterUuid = types.StringValue(r.ClusterAgentTokenCreate.ClusterAgentToken.Cluster.Uuid)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (ct *ClusterAgentToken) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state ClusterAgentTokenResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	log.Printf("Getting cluster agent tokens for cluster %s ...", state.ClusterUuid.ValueString())
+	tokens, err := getClusterAgentTokens(ct.client.genqlient, ct.client.organization, state.ClusterUuid.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read Cluster Agent Tokens",
+			fmt.Sprintf("Unable to read Cluster Agent Tokens: %s", err.Error()),
+		)
+		return
+	}
+
+	for _, edge := range tokens.Organization.Cluster.AgentTokens.Edges {
+		if edge.Node.Id == state.Id.ValueString() {
+			log.Printf("Found cluster Token with Description %s in cluster %s", edge.Node.Id, state.ClusterUuid.ValueString())
+			state.Description = types.StringValue(edge.Node.Description)
+			resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+			return
+		}
+	}
+
+}
+
+func (ct *ClusterAgentToken) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state, plan ClusterAgentTokenResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	_, err := updateClusterAgentToken(
```

Comment:
I think adding another `log.Printf` to the Update fn like the other operations might be good for consistency 👍

```suggestion
    log.Printf("Updating cluster token %s", state.Id.ValueString())
	_, err := updateClusterAgentToken(
```